filters/auth/oidc_introspection.go

```diff
@@ -173,6 +173,18 @@ func (p pathQuery) String() string {
 	return fmt.Sprintf("path: '%s*', matching: %s", p.path, strings.Join(p.queries, " ,"))
 }
 
+// Splits space-delimited GJSON queries ignoring spaces within quoted strings
+func splitQueries(s string) (q []string) {
+	for _, p := range strings.Split(s, " ") {
+		if len(q) == 0 || strings.Count(q[len(q)-1], `"`)%2 == 0 {
+			q = append(q, p)
+		} else {
+			q[len(q)-1] = q[len(q)-1] + " " + p
+		}
+	}
+	return
+}
+
```

Comment:
It is better then simple `strings.Split(s, " ")` but will not work properly with escaped quote within string, e.g. `/asdf/:groups.#(%"wh\"ite space")`

Reply:
The split by whitespace could be avoided if GJSON would support OR conditions which it does not https://github.com/tidwall/gjson/issues/162
We could also allow having the same path multiple times in the filter list to emulate OR, unfortunately whitespace-separated queries for the same path were chosen and probably we should stick to it for backward compatibility.

Alternative implementation could use query validation but there is no such utility https://github.com/tidwall/gjson/issues/180

We could adjust quote count for the number of escaped quotes like
```go
last := q[len(q)-1]
unescapedQuotes := strings.Count(last, `"`) - strings.Count(last, `\"`)
```
but it seems GJSON does not handle quotes properly https://github.com/tidwall/gjson/issues/215 so maybe this simplistic approach is good enough until the need for the values with quotes arises.

Reply:
> We could adjust quote count for the number of escaped quotes

This is also an ad-hoc as it would fail on escaped backslash like e.g. in `/asdf/:groups.#(%"wh\\") groups.#(%"whatever")`

https://github.com/tidwall/gjson/issues/215 was fixed in the meantime

Reply:
In the meantime https://github.com/tidwall/gjson/issues/215 was fixed so it supports escaped sequences in the query now.

With that an alternative workaround may be to update library and use escape sequence for the whitespace like `oidcClaimsQuery("/login:groups.#[==\"appX\\u0020Tester\"]")` which could be tested in the playground https://gjson.dev/ with `groups.#[=="appX\u0020Tester"]` query and 
```json
{
  "groups": ["X", "appX Tester"]
}
```
document

Reply:
this looks good to me, and the workarounds using whitespace codes is also fine. I am thinking though the case about escaped quote signs. That would probably require iterating through the runes instead of using strings.Split. Let me check how that would look

Reply:
it would look something like this:

```Go
// Splits space-delimited GJSON queries ignoring spaces within quoted strings
// and applies trimming.
func splitQueries(s string) (q []string) {
	var (
		query           []rune
		quoted, escaped bool
	)

	for _, r := range []rune(s) {
		if escaped {
			escaped = false
			query = append(query, r)
			continue
		}

		if quoted {
			switch r {
			case '\\':
				escaped = true
			case '"':
				quoted = false
			}

			query = append(query, r)
			continue
		}

		if len(query) > 0 {
			if unicode.IsSpace(r) {
				q = append(q, string(query))
				query = nil
				continue
			}

			if r == '"' {
				quoted = true
			}

			query = append(query, r)
			continue
		}

		if unicode.IsSpace(r) {
			continue
		}

		if r == '"' {
			quoted = true
		}

		query = append(query, r)
	}

	if len(query) > 0 {
		q = append(q, string(query))
	}

	return q
}
```

Reply:
It is interesting that space is permitted within array selector outside of the string literal, i.e. `groups.#[ == "X" ]` is a valid query that will stumble upon out split code both existing and proposed.

Reply:
valid point :+1: